packages/studio/src/schemas/page.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from "zod";
+
+export const getEditPageSchema = z.object({
+  pageId: z.string(),
+  siteId: z.string()
+})
+
+export const updatePageSchema = z.object({
+  // NOTE: We allow both to be empty now, 
+  // in which case this is a no-op. 
+  // We are ok w/ this because it doesn't 
+  // incur any db writes
+  parentId: z.string().nullable(),
+  pageName: z.string().nullable(),
+  siteId: z.string(),
+  pageId: z.string()
+})
+
+export const updatePageBlobSchema = z.object({
+  pageId: z.string(),
+  siteId: z.string(),
+  content: z.string()
+})
+
+export const createPageSchema = z.object({
+  pageName: z.string(),
+  pageTitle: z.string(),
+  // TODO: add the actual layouts in here
+  layout: z.enum(["content"]).default("content"),
+  siteId: z.string(),
+  // NOTE: implies that top level pages are allowed
+  folderId: z.string().nullable()
```

Comment:
hmm so we are locking down to two states at the top level ya, one is top level page another is top level folder ya?

Reply:
for now, yes. underlying schema will expose only 1 table still - i think this distinction makes sense because pages/folders are what the frontend sees (and queries), but the internals are db-specific and limited to a single table. 

this also implies that there might be some logical differences between a page-level route/folder-level route, such as (but not limited to) validation of names